microprofile/metrics/src/main/java/io/helidon/microprofile/metrics/MetricsCdiExtension.java

```diff
@@ -643,53 +726,51 @@ private static boolean chooseRestEndpointsSetting(Config metricsConfig) {
         return MetricType.from(clazz == null ? metric.getClass() : clazz);
     }
 
-    private void recordAnnotatedGaugeSite(@Observes @WithAnnotations(Gauge.class) ProcessAnnotatedType<?> pat) {
-        LOGGER.log(Level.FINE, () -> "recordAnnoatedGaugeSite for class " + pat.getAnnotatedType().getJavaClass());
-        AnnotatedType<?> type = pat.getAnnotatedType();
+    private void recordAnnotatedGaugeSite(@Observes ProcessManagedBean<?> pmb) {
+        AnnotatedType<?> type = pmb.getAnnotatedBeanClass();
+        Class<?> clazz = type.getJavaClass();
 
-        LOGGER.log(Level.FINE, () -> "Processing annotations for " + type.getJavaClass().getName());
+        LOGGER.log(Level.FINE, () -> "recordAnnoatedGaugeSite for class " + clazz);
+        LOGGER.log(Level.FINE, () -> "Processing annotations for " + clazz.getName());
 
         // Register metrics based on annotations
-        AnnotatedTypeConfigurator<?> configurator = pat.configureAnnotatedType();
-        Class<?> clazz = configurator.getAnnotated().getJavaClass();
-
         // If abstract class, then handled by concrete subclasses
         if (Modifier.isAbstract(clazz.getModifiers())) {
             return;
         }
 
-        Annotation annotation = type.getAnnotation(RequestScoped.class);
-        if (annotation != null) {
-            errors.fatal(clazz, "Cannot configure @Gauge on a request scoped bean");
-            return;
-        }
-
-        if (type.getAnnotation(ApplicationScoped.class) == null && type.getAnnotation(Singleton.class) == null) {
-            if (ConfigProvider.getConfig().getOptionalValue("metrics.warn-dependent", Boolean.class).orElse(true)) {
-                LOGGER.warning("@Gauge is configured on a bean " + clazz.getName()
-                                       + " that is neither ApplicationScoped nor Singleton. This is most likely a bug."
-                                       + " You may set 'metrics.warn-dependent' configuration option to 'false' to remove "
-                                       + "this warning.");
+        // Process @Gauge methods keeping non-private declared on this class
+        for (AnnotatedMethod method : type.getMethods()) {
+            Method javaMethod = method.getJavaMember();
+            if (!javaMethod.getDeclaringClass().equals(clazz)
+                    || Modifier.isPrivate(javaMethod.getModifiers())
+                    || !method.isAnnotationPresent(Gauge.class)) {
+                continue;
             }
+            Annotation requestScopedAnnotation = type.getAnnotation(RequestScoped.class);
```

Comment:
Better to use `pmb.getBean().getScope()` unless I'm missing something.

Reply:
Done.